### PR TITLE
Clarify that OTEL_SEMCONV_STABILITY_OPT_IN is a comma-separated list of values

### LINKED ISSUE #3488: HTTP SemConv transition plan is not extensible
The HTTP SemConv transition plan defines an environment variable with potential values `http`, `http/dup`, etc. It is my understanding that the plans for future semconv areas like messaging will add additional possible values to this environment variable ( e.g. `OTEL_SEMCONV_STABILITY_OPT_IN =http/dup/messaging`) [source](https://github.com/open-telemetry/opentelemetry-specification/pull/3443#issuecomment-1536544773).

After the transition plan merged, it was pointed out to me by @Flarna that the variable requires a predefined method for multiple values such as comma-separated values. Otherwise an implementation which simply checks `env.OTEL_SEMCONV_STABILITY_OPT_IN == "http"` or uses a switch/case will be broken in the case that multiple values exist as in the example above.

Action items:
- Document in specification/configuration/sdk-environment-variables.md
- Add explicit comma separated format
  - Alternatively: explicitly decide we are not going to extend it in the future

/cc @trask

### PR DESCRIPTION
**Pull request by trask**
Fixes #3488

## Changes

Clarifies that the `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable is a comma-separated list of values

### CODE REVIEW

**Review by reyang — APPROVED** [review]

**Review by pyohannes — APPROVED** [review]

File: specification/metrics/semantic_conventions/http-metrics.md [hunk 1]
```diff
@@ -36,15 +36,16 @@ operations. By adding HTTP attributes to metric events it allows for finely tune
 >   include stabilization of a core set of networking attributes which are also used
 >   in HTTP instrumentations).
 > * SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN`
```
Comment (arminru) [inline specification/metrics/semantic_conventions/http-metrics.md:38]: I think the env var should also be listed in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md - either in this PR or a follow-up.

**Review by arminru — APPROVED** [review]

File: specification/metrics/semantic_conventions/http-metrics.md [hunk 2]
```diff
@@ -36,15 +36,16 @@ operations. By adding HTTP attributes to metric events it allows for finely tune
...
->   in the existing major version which supports the following values:
->   * `none` - continue emitting whatever version of the old experimental
->     HTTP and networking attributes the instrumentation was emitting previously.
->     This is the default value.
+>   in the existing major version which is a comma-separated list of values.
```
Comment (dyladan) [inline specification/metrics/semantic_conventions/http-metrics.md:39]:
```suggestion
>   in the existing major version which is a case-insensitive, comma-separated list of values with optional whitespace.
```

**Review by carlosalberto — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Review by jsuereth — CHANGES_REQUESTED** [review]
This PR needs to move to the new [Semantic Conventions](https://github.com/open-telemetry/semantic-conventions) repository.

File: specification/metrics/semantic_conventions/http-metrics.md (hunk 2 above)
Comment (dyladan) [reply specification/metrics/semantic_conventions/http-metrics.md:39]: ok fine with me

File: specification/metrics/semantic_conventions/http-metrics.md (hunk 1 above)
Comment (arminru) [reply specification/metrics/semantic_conventions/http-metrics.md:38]: In lieu of a better place (that's the first instrumentation config env var we defined in the spec) I think it should be OK to add it there for now, so all spec-defined env vars are listed in one place.

**Pull request closed without merging**